github/resource_github_branch_protection_v3.go

```diff
@@ -53,12 +53,30 @@ func resourceGithubBranchProtectionV3() *schema.Resource {
 							Default:  false,
 						},
 						"contexts": {
-							Type:     schema.TypeSet,
-							Optional: true,
+							Type:       schema.TypeSet,
+							Optional:   true,
+							Deprecated: "GitHub is deprecating the use of `contexts`. Use `check` blocks instead.",
 							Elem: &schema.Schema{
 								Type: schema.TypeString,
 							},
 						},
+						"check": {
+							Type:     schema.TypeList,
+							Optional: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"context": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+									"app_id": {
+										Type:     schema.TypeInt,
+										Optional: true,
+										Default:  -1,
```

Comment:
Upon reflection this should be nullable.
The API states the field can be _ommitted_ to allow the most recent app, **or** set to _-1_ to allow any app:
https://docs.github.com/en/rest/branches/branch-protection?apiVersion=2022-11-28#update-branch-protection
https://pkg.go.dev/github.com/google/go-github/v48@v48.1.0/github#RequiredStatusCheck